### PR TITLE
uuid for scrape requests 

### PR DESCRIPTION
**Pull request by glaslos**
Replaced the pseudo id with UUIDv4. Added a HMAC signature of the UUID which allows the verification of a UUID. A secret for HMAC is generated during runtime and doesn't not persist between restarts.

### CODE REVIEW

**Comment by brian-brazil** [comment]
What's the attack vector you're looking to protect against?

**Comment by glaslos** [comment]
@brian-brazil apparently you think spoofing could be an issue :wink: https://github.com/RobustPerception/PushProx/blob/master/proxy/coordinator.go#L65-L66

**Comment by brian-brazil** [comment]
Hmm, just a UUID is probably enough then. If you can get to the HTTP server there's plenty of ways to do a DOS.

**Comment by glaslos** [comment]
Makes sense, I removed the signing :+1:

File: proxy/coordinator.go [hunk 1]
```diff
@@ -43,28 +44,33 @@ type Coordinator struct {
...
 		known:     map[string]time.Time{},
 		logger:    logger,
 	}
+
+	// generating a random byte array to sign the request IDs
+	signKey := make([]byte, 12)
+	if _, err := rand.Read(signKey); err != nil {
+		return nil, err
+	}
+	c.signKey = signKey
```
Comment (brian-brazil) [inline proxy/coordinator.go]: This can go

Comment (glaslos) [reply proxy/coordinator.go]: Good catch! Done.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**